### PR TITLE
fix(index): remove empty assets

### LINKED ISSUE #738: [4.0.0-beta.0] emitting empty CSS files when using dynamic import
**General:**

When using dynamic `import()` code-splitting, ETWP creates an empty (0 byte) CSS file for the main entry point (e.g. `main.css`) even if no CSS was imported. Additionally, if using `optimization.splitChunks.chunks: "all"` to split initial chunks in addition to async, it creates an empty CSS file for vendors as well (e.g. `vendors~main.css`) even if no vendor (`node_modules`) CSS was imported. I'm assuming these two cases are the same root problem, I just don't know enough about how it works to know for sure.

**Real-world use case:**

I include CSS from `node_modules` in my entry point, which gets split into `vendors~main.css` as expected, but ETWP also creates an empty `main.css` when I don't have any local CSS imports.

**Reproduction repo:**

https://github.com/wKovacs64/etwp-empty-files-demo

**Versions:**

extract-text-webpack-plugin: 4.0.0-beta.0
webpack: 4.0.1

### PR DESCRIPTION
**Pull request by lili21**
### `Issues`

- Fixes #738

### CODE REVIEW

**Comment by lili21** [comment]
Hello, anyone please look at this ?

**Comment by evilebottnawi** [comment]
@lili21 can you add tests?

File: commitlint.config.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const Configuration = {
     'subject-full-stop': [2, 'never', '.'],
     'type-case': [2, 'always', 'lower-case'],
     'type-empty': [2, 'never'],
-    'type-enum': [
+    'type-enum': [2, 'always', [
```
Comment (evilebottnawi) [inline commitlint.config.js]: Also don't change eslint config file

Comment (lili21) [reply commitlint.config.js]: I think this lint config is invalid. cause commitlint keep yelling at me about value of `type-enum` is wrong.

Comment (evilebottnawi) [reply commitlint.config.js]: @lili21 this PR please send to `webpack-default` repo

Comment (lili21) [reply commitlint.config.js]: looks like `webpack-default` have nothing to do with `commitlint.config.js`

Comment (evilebottnawi) [reply commitlint.config.js]: @lili21 https://github.com/webpack-contrib/webpack-defaults/tree/next/templates

**Comment by lili21** [comment]
I guess I have to change the intergration test code to test empty chunk rather than just add test case.
any better ideal ?

File: commitlint.config.js (hunk 1 above)
Comment (lili21) [reply commitlint.config.js]: Got it.

**Comment by evilebottnawi** [comment]
@lili21 Yep add tests case https://github.com/webpack-contrib/extract-text-webpack-plugin/tree/master/test/cases

**Comment by lili21** [comment]
what I mean is I can't just add test case. I have to change the `webpack-integration.test.js` code for testing if empty chunk have been generated.

**Comment by lili21** [comment]
couldn't find a way to test it without change the way of integration test. maybe you can give me some advice ?

File: test/cases/empty-chunk/a.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+require('./a.txt');
+require('./b.txt');
```
Comment (evilebottnawi) [inline test/cases/empty-chunk/a.js]: Add `\n` in end of file :+1:

File: test/webpack-integration.test.js [hunk 3]
```diff
@@ -47,24 +48,20 @@ describe('Webpack Integration Tests', () => {
...
-        fs.readdirSync(expectedDirectory).forEach((file) => {
-          const filePath = path.join(expectedDirectory, file);
-          const actualPath = path.join(outputDirectory, file);
-          expect(readFileOrEmpty(actualPath)).toEqual(
-            readFileOrEmpty(filePath)
-          );
-          expect(readFileOrEmpty(actualPath)).toMatchSnapshot();
-        });
+
+        glob
```
Comment (evilebottnawi) [inline test/webpack-integration.test.js]: Why need glob here?

Comment (lili21) [reply test/webpack-integration.test.js]: I need to get all files in a directory recursively. I can do it without glob, but why not ?

Comment (evilebottnawi) [reply test/webpack-integration.test.js]: :+1:

File: test/cases/empty-chunk/a.js (hunk 2 above)
Comment (lili21) [reply test/cases/empty-chunk/a.js]: 👌

**Comment by lili21** [comment]
@evilebottnawi

**Comment by wKovacs64** [comment]
I see the "just use `mini-css-extract-plugin`" comments closing other issues, but since this PR was already merged into `next`, will we see a new release that includes it?

**Comment by michael-ciniawsky** [comment]
Yep, I will cut a 'final' release including this fix. I just need to finish some other chores before and add a deprecation note to the README and `npm deprecate` the package properly. Hopefully today...

**Pull request merged**